### DISCUSSION TITLE
An official I18N integration.

### DISCUSSION THREAD

**Discussion by arstnei0** [withastro/roadmap#322] [Proposal]
Does anyone have a idea about creating an official I18N integration in Astro?

I believe that it is easy to make one. I do need one for my project.

Think about some great features: 

- Support UI strings
- Support pages translation

**Comment by L1lith** [comment]
> Does anyone have a idea about creating an official I18N integration in Astro?
> 
> I believe that it is easy to make one. I do need one for my project.
> 
> Think about some great features:
> 
>     * Support UI strings
> 
>     * Support pages translation

I would like to see improvements in the translation area but tbh I think [astro-i18next](https://github.com/yassinedoghri/astro-i18next) is the best solution I've seen. Perhaps there could be some changes made to Astro to help make this library even better? :)

Edit: Also see [this list of awesome features they're trying to implement](https://github.com/yassinedoghri/astro-i18next/issues/19)

**Comment by matthewp** [comment]
We added i18n support in Astro 4! Closing.
